### PR TITLE
Add connector for WDR

### PR DESCRIPTION
**Pull request by Ceddicedced**
I created a connector for the wdr stations. I used this new connector today while listening to 1Live and couldn't experience any errors.

**Describe the changes you made**
Added a connector for the WDR radio stations (1Live, WDR2, WDR3, WDR4, WDR5, COSMO, etc.)

I faced some problems with the station WDR5 because the track information was like this: "WDR 5 Quarks - Wissenschaft und mehr mit Sebastian Sonntag". But I were not able to filter strings like this with the following code:
`
Connector.isScrobblingAllowed = () => {
	const artistTrack = Util.getTextFromSelectors(
		Connector.artistTrackSelector,
	);
	const disallowedStrings = [
		'1Live',
		'WDR2',
		'WDR3',
		'WDR4',
		'WDR5',
		'COSMO',
	];

	// Check if the artistTrack includes any of the disallowed strings
	const containsDisallowedString =
		artistTrack !== null &&
		disallowedStrings.some((disallowedString) =>
			artistTrack.includes(disallowedString),
		);

	// Scrobble only if none of the disallowed strings are included
	return !containsDisallowedString;
};
`

Maybe there is a nicer solution...

### CODE REVIEW

**Comment by yayuyokitano** [comment]
Thank you for the contribution!

You can run Connector.getArtist() in the isScrobblingAllowed function btw if that makes for a cleaner solution than running the Util.getTextFromSelectors() raw.

As for the WDR5 issue, to me it looks like the issue is you need to add a space to "WDR 5" in the disallowedStrings array.

Also, it looks like you may not have submitted the full script in your PR, please have a look and confirm, it looks a bit interrupted.

**Comment by Ceddicedced** [comment]
Thank you so much for the quick feedback!

Thanks for pointing out the string issues, such a rookie mistake by me...

Should look better now!

File: src/connectors/wdr.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+export {};
+
+// Connector for the WDR radio stations (1Live, WDR2, WDR3, WDR4, WDR5, COSMO, etc.)
+
+Connector.playerSelector = '.wdrrCurrentChannels';
+
+Connector.artistTrackSelector = '.wdrrCurrentShowTitleTitle';
+
+Connector.isPlaying = () => {
```
Comment (yayuyokitano) [inline src/connectors/wdr.ts:9]:
```suggestion
Connector.pauseButtonSelector = '#playCtrl.playing';
```
Does this work instead?

**Review by yayuyokitano — COMMENTED** [review]
Overall looks good, just one suggestion to get playing detection more in line with our best practices, if it works.

File: src/connectors/wdr.ts (hunk 1 above)
Comment (Ceddicedced) [reply src/connectors/wdr.ts:13]: Yes this is working perfectly fine. Thank you

File: src/connectors/wdr.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+});
+
+Connector.playerSelector = '.wdrrCurrentChannels';
+
+Connector.artistTrackSelector = '.wdrrCurrentShowTitleTitle';
+
+Connector.pauseButtonSelector = '#playCtrl.playing';
+	const playButton = document.querySelector('#playCtrl');
+	return playButton !== null && playButton.classList.contains('playing');
+};
```
Comment (yayuyokitano) [inline src/connectors/wdr.ts:16]:
```suggestion
```

My apologies, bad suggestion from my end, this also needs gone

File: src/connectors/wdr.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		artistTrack !== null &&
+		disallowedStrings.some((disallowedString) =>
+			artistTrack.includes(disallowedString),
+		);
+
+	// Scrobble only if none of the disallowed strings are included
+	return !containsDisallowedString;
+};
+
+function cleanupArtist(artist: any) {
```
Comment (yayuyokitano) [inline src/connectors/wdr.ts:39]:
```suggestion
function cleanupArtist(artist: string) {
```
I believe this will fix the linter errors

**Review by yayuyokitano — APPROVED** [review]
LGTM. Thank you for the contribution!

**Pull request merged**